### PR TITLE
Add controllers and route

### LINKED ISSUE #6: [xpt 3.0]Add controllers and routes
Add controllers for user, motor, and reservations

### PR DESCRIPTION
**Pull request by binodbhusal**
**The following tasks were accomplished in this milestone**

- [x] Added Motors,Users, Reservations and Stores_location controllers
- [x] Added routes 
- [x] Added seed data
- [x] Installed Gem rake-cors and will_paginate

### CODE REVIEW

File: app/controllers/api/motors_controller.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
+class Api::MotorsController < ApplicationController
+  before_action :set_motor, only: %i[show update destroy]
+
+  # GET /motors
+  def random
+    @motors = Motor.all
+    @motors = @motors.where(brand_name: params[:brand_name]) if params[:brand_name].present?
+    @motors = @motors.paginate(page: params[:page], per_page: params[:per_page] || 10)
+    render json: @motors
+  end
```
Comment (rahimialijan) [inline app/controllers/api/motors_controller.rb]: It would be nice if you use 'index' instead of 'random' 😨 Kindly fix it 👍

**Review by rahimialijan — CHANGES_REQUESTED** [review]
Hi @binodbhusal 

# Required Changes
Good job on implementing some of the requirements, There are some issues that you still need to work on to go to the next project but you are almost there! 😊


## Highlights 🥇
✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Gitflow
✔️ Following some requirements

**Comment by dieum512** [comment]
Hey @binodbhusal 

Your project is complete! There is nothing else to say other than... it's time to merge it :ok_hand: 
Congratulations! :tada: 



### HIGHLIGHTS
- [x] No  linter errors :heavy_check_mark: 
- [x] Good Commit messages :heavy_check_mark: 
- [x] Repository documented in a professional way :heavy_check_mark:

**Review by dieum512 — APPROVED** [review]
Hey @binodbhusal 

Your project is complete! There is nothing else to say other than... it's time to merge it :ok_hand: 
Congratulations! :tada: 



### HIGHLIGHTS
- [x] No  linter errors :heavy_check_mark: 
- [x] Good Commit messages :heavy_check_mark: 
- [x] Repository documented in a professional way :heavy_check_mark:

**Pull request merged**